src/lib.rs

```diff
@@ -268,6 +271,20 @@ where
   }
 }
 
+/// A resource sink for [`RecursiveSNARK`]
```

Comment:
I don't think "resource sink" is known vocabulary you can expect the developer to know. 

In fact, while the elements contained here are the destination of a particular iterations of computation of a `foo_into` function, they are not the final destination of the computed data, and so I'm not sure if the word [*sink*](https://en.wikipedia.org/wiki/Sink_(computing)) is appropriate.

I'd like to see a short comment stating what this is used for without pre-requisites on the part of the reader: storing scratch values computed in a tight loop, which allows reusing the corresponding memory allocations.

Reply:
I renamed to `ResourceBuffer`, which should be more appropriate